pre_commit/languages/docker.py

```diff
@@ -77,7 +78,18 @@ def install_environment(
 
 
 def get_docker_user() -> Tuple[str, ...]:  # pragma: win32 no cover
+    output = subprocess.check_output(
+        ('docker', 'system', 'info'),
+        text=True,
+    )
     try:
+        for line in output.splitlines():
+            # rootless docker has "rootless"
+            # rootless podman has "rootless: true"
+            if line.strip().startswith('rootless'):
+                if 'false' not in line:
+                    return ()  # no -u for rootless
+                break
```

Comment:
Not just output, move this whole thing out.

Reply:
Done. Thanks @webknjaz!